src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildData.java

```diff
@@ -135,8 +135,32 @@ public class BuildData implements Serializable {
     private Long startTime;
     private Long endTime;
     private Long duration;
-    private Long millisInQueue;
-    private Long propagatedMillisInQueue;
+
+    /**
+     * Queue time that is reported by the Jenkins Queue API.
+     * This is the time that the build was sitting in queue BEFORE starting to execute:
+     * <ul>
+     *  <li>if it was waiting for another build to finish because of the parallel execution settings</li>
+     *  <li>if it was waiting for a "quiet period" to finish because of the throttle settings</li>
+     *  <li>if it was waiting for an executor to become available (this point only applies to Freestyle builds; a pipeline waiting for executor will report {@link #propagatedMillisInQueue}</li>
+     * </ul>
+     * <p>
+     * This queue time goes BEFORE timestamp that is reported by {@link Run#getStartTimeInMillis()}
+     * and is NOT included into the duration reported by {@link Run#getDuration()}.
+     */
+    private long millisInQueue;
+
+    /**
+     * Queue time that is propagated from a child node of a pipeline:
+     * if a pipeline is configured to execute in a specific executor (with the top-level `agent {}` section),
+     * it will be reported as started even if that executor is not available.
+     * In this case the time spent waiting for the executor will be reported by the first child node of the pipeline,
+     * and propagated to the pipeline from there.
+     * <p>
+     * This queue time goes AFTER timestamp that is reported by {@link Run#getStartTimeInMillis()}
+     * and is INCLUDED into the duration reported by {@link Run#getDuration()}.
+     */
+    private long propagatedMillisInQueue;
```

Comment:
Do we know how does this work when the pipeline uses multiple agents? 
https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-multiple-agents

Reply:
As `BuildData` represent the entire build, only time spent waiting for the agent specified in the top-level `agent {}` section is propagated, which I think is the correct behaviour. 
For individual stages the time spent waiting for their respective agents will be accounted for in the `PipelineStepData` instances